### PR TITLE
test: [CDS-69669]: Add tests for StepCommands, StepGroupStep and fix edge cases

### PR DESCRIPTION
**Pull request by rameshx**
### Summary

This fixes minor bugs and adds jest tests for the following:

1. rendering and submitting steps, step groups, and template steps
2. switching tabs for steps containing multiple tabs
3. Save as Template use cases for steps and step groups
4. Ensuring values from both tabs are picked
5. `StepGroupStep`

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineStudio/StepCommands/__tests__/__snapshots__/StepCommands.test.tsx.snap
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/StepCommands/__tests__/__snapshots__/StepCommands.test.tsx.snap:1]: removed all snapshots as they are not helpful anymore

File: src/modules/70-pipeline/components/PipelineStudio/StepCommands/StepCommands.tsx [hunk 1]
```diff
@@ -299,9 +302,6 @@ export function StepCommands(
                     onChange={values => {
                       onChange?.(getValues({ advancedValues: values }))
                     }}
-                    onUpdate={values => {
-                      onUpdate?.(getValues({ advancedValues: values }))
-                    }}
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/StepCommands/StepCommands.tsx:304]: removed `onUpdate` prop as it is not used anywhere within `AdvancedSteps` component

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]
